fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashActivity.java

```diff
@@ -98,80 +83,81 @@ public static DigitalCashViewModel obtainViewModel(FragmentActivity activity) {
     return new ViewModelProvider(activity).get(DigitalCashViewModel.class);
   }
 
-  public void setupNavigationBar() {
-    navbar = findViewById(R.id.digital_cash_nav_bar);
-    navbar.setOnItemSelectedListener(
-        item -> {
-          int id = item.getItemId();
-          if (id != mViewModel.getCurrentSelectedItem().getValue()) {
-            // This prevents the update to be done multiple times. It is done here rather than
-            // in viewModel because otherwise this would be executed twice
-            mViewModel.setCurrentSelectedItem(id);
-          } else {
-            if (id == R.id.digital_cash_home_menu) {
-              Log.d(TAG, "Opening home fragment");
-              openHomeFragment();
-            } else if (id == R.id.digital_cash_history_menu) {
-              Log.d(TAG, "Opening history fragment");
-              openHistoryFragment();
-            } else if (id == R.id.digital_cash_send_menu) {
-              Log.d(TAG, "Opening send fragment");
-              openSendFragment();
-            } else if (id == R.id.digital_cash_receive_menu) {
-              Log.d(TAG, "Opening receive fragment");
-              openReceiveFragment();
-            } else if (id == R.id.digital_cash_issue_menu) {
-              Log.d(TAG, "Trying to open issue fragment");
-              handleOpenIssue();
-            }
-          }
-          return true;
-        });
-  }
-
-  private void openHomeFragment() {
+  @Override
+  protected DigitalCashTab findTabByMenu(int menuId) {
+    return DigitalCashTab.findByMenu(menuId);
+  }
+
+  @Override
+  protected boolean openTab(DigitalCashTab tab) {
+    switch (tab) {
```

Comment:
it might be good to keep the `Log.d`, maybe just one logging the `tab` value though, as opposed to one per case.

Reply:
I moved the log of the tab directly in the navigation. So in the navigationActivity at lines 36, 39 and 42.